### PR TITLE
helm repository access method

### LINKED ISSUE #352: Create access type "helmRepository" for artifact type "helmChart"
**What would you like to be added**:
There should be an access method "helmRepository" that allows to specify the location of the artifact in a given Helm repository

**Why is this needed**:
Currently helmCharts need to be available as OCI artifact or tar, but the standard way of consuming Helm charts is to store them as packaged as tgz in a Helm repository.

### PR DESCRIPTION
**Pull request by mandelsoft**
**What this PR does / why we need it**:

This PR provides access to helm charts in helm repositories:
- The helm input type is extended to also refer to charts provided in helm repositories in addition to charts in the filesystem.

  In general, helm charts provided by this input type are always stored as local blobs using the OCI based format, regardless
  of the storage technologies used by the chart source. Because of an existing uploader for OCI artifacts, transporting those 
  resources by value will provide remotely accessible helm charts using OCI-based helm references for the helm tooling.

- The helm access method can be used for OCM resources to refer to charts in helm repositories

Attention: MimeTypes for helm chart blobs are dependent on the access method:
- OCI based:  `application/vnd.oci.image.manifest.v1+tar+gzip`
- Archive based: `application/vnd.cncf.helm.chart.content.v1.tar+gzip`

Helm chart references used by the helm repository access method are plain tgz archives as provided by
the helm repository. There is no uploader, so far, which is capable to upload those helm charts to a repository, again,
 once such an artifact is transported by value, neither into a helm repository nor into an OCI registry.

If this is added later, it is important to be aware of the digest handling. The digests of helm repository charts and OCI repository based charts are calculated in different ways: OCI manifest digest vs binary blob digest of the archive.

If conversions should be handled by a helm chart uploader capable to upload archive based helm charts to OCI repositories (and vice versa) there must also be an explicit helm *digester*, which provides MimeType independent digests for artifacts of type `helmChart` to avoid breaking signatures.

The online docu feature has been extended to provide more information for the credential handling for dedicated
consumer types. For example helm repositories now require different credential attributes than other consumers.
The consumer now provides info about the evaluated credential attributes, as well as the used identity attributes.
The online-help now also includes this information to tell the user what properties he has to use for an intended consumer.
All the already supported credential consumers have been adapted accordingly. To align the credential handling
a uniform package structure has been established for all consumer types.

**Which issue(s) this PR fixes**:
Fixes #352 #218

**Special notes for your reviewer**:

**Release Notes**:

```feature user

```

### CODE REVIEW

**Review by jensh007 — APPROVED** [review]

**Review by jensh007 — DISMISSED** [review]

File: cmds/ocm/topics/common/credentials/topic.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+This is achieved by establishing a credential request mechanism based on
+generic consumer identities and credential property sets.
+On the one hand every kind of credential consumer uses a dedicated consumer
+type (string). Additionally, it defines a set of properties further describing
+the target/context credentials are required for.
+
+On the other hand credentials can be defined for such sets of identities
+with partial sets of properties (see <CMD>ocm configfile</CMD>). A credential
+request is then matched again the available credential settings using matchers,
```
Comment (fabianburth) [inline cmds/ocm/topics/common/credentials/topic.go:43]:
```suggestion
request is then matched against the available credential settings using matchers,
```

File: cmds/ocm/topics/common/credentials/topic.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+which might by specific for dedicated kinds of requests. For example, a hostpath
```
Comment (fabianburth) [inline cmds/ocm/topics/common/credentials/topic.go:44]:
```suggestion
which might be specific for dedicated kinds of requests. For example, a hostpath
```

File: docs/reference/ocm_credential-handling.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+This is achieved by establishing a credential request mechanism based on
+generic consumer identities and credential property sets.
+On the one hand every kind of credential consumer uses a dedicated consumer
+type (string). Additionally, it defines a set of properties further describing
+the target/context credentials are required for.
+
+On the other hand credentials can be defined for such sets of identities
+with partial sets of properties (see [ocm configfile](ocm_configfile.md)). A credential
+request is then matched again the available credential settings using matchers,
```
Comment (fabianburth) [inline docs/reference/ocm_credential-handling.md:19]:
```suggestion
request is then matched against the available credential settings using matchers,
```

File: docs/reference/ocm_credential-handling.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+which might by specific for dedicated kinds of requests. For example, a hostpath
```
Comment (fabianburth) [inline docs/reference/ocm_credential-handling.md:20]:
```suggestion
which might be specific for dedicated kinds of requests. For example, a hostpath
```

**Review by jensh007 — DISMISSED** [review]

File: pkg/contexts/credentials/builtin/github/identity/identity.go
Comment (fabianburth) [inline pkg/contexts/credentials/builtin/github/identity/identity.go:1]: Why is this file here and not under pkg/contexts/credentials/identity/github (parallel to hostpath?)

File: pkg/contexts/ocm/accessmethods/helm/method.go [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+// New creates a new Helm Chart accessor for helm repositories.
+func New(chart string, repourl string) *AccessSpec {
+	return &AccessSpec{
+		ObjectVersionedType: runtime.NewVersionedObjectType(Type),
+		HelmChart:           chart,
+		HelmRepository:      repourl,
+	}
+}
+
+// AccessSpec describes the access for a oci registry.
```
Comment (fabianburth) [inline pkg/contexts/ocm/accessmethods/helm/method.go:43]:
```suggestion
// AccessSpec describes the access for a specific version of a helm chart within a helm repository.
```

**Review by jensh007 — APPROVED** [review]
